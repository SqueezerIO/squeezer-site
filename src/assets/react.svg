<svg viewBox="0 0 47 42" xmlns="http://www.w3.org/2000/svg">
  <title>
    react js logo
  </title>
  <g transform="translate(1)" fill="none" fill-rule="evenodd">
    <path d="M30.5 35.454c-3.13 1.808-10.293-3.827-15.562-12.954-5.27-9.127-6.57-18.147-3.438-19.954C14.63.738 21.793 6.373 27.062 15.5c5.27 9.127 6.57 18.147 3.438 19.954zm1 1.733c4.522-2.61 3.07-12.682-2.706-22.687S15.022-1.797 10.5.813C5.978 3.423 7.43 13.495 13.206 23.5S26.978 39.797 31.5 37.187z" fill="#744C9E"/>
    <path d="M0 19c0 5.22 9.448 9 21 9s21-3.78 21-9c0-5.22-9.448-9-21-9S0 13.78 0 19zm2 0c0-3.616 8.46-7 19-7s19 3.384 19 7-8.46 7-19 7-19-3.384-19-7z" fill="#744C9E"/>
    <path d="M28.794 23.5C34.57 13.495 36.022 3.424 31.5.813 26.978-1.797 18.982 4.495 13.206 14.5 7.43 24.505 5.978 34.576 10.5 37.187c4.522 2.61 12.518-3.682 18.294-13.687zm-13.856-8C20.208 6.373 27.368.738 30.5 2.546c3.13 1.807 1.832 10.827-3.438 19.954-5.27 9.127-12.43 14.762-15.562 12.954-3.13-1.807-1.832-10.827 3.438-19.954z" fill="#744C9E"/>
    <rect fill="#ffb238" x="17" y="15" width="8" height="8" rx="4"/>
  </g>
</svg>
