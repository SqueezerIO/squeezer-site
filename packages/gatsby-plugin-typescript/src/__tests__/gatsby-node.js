jest.mock(`path`, () => {
  return {
    resolve: () => ``,
  }
})
const {
  resolvableExtensions,
  modifyWebpackConfig,
  preprocessSource,
} = require(`../gatsby-node`)

describe(`gatsby-plugin-typescript`, () => {
  it(`returns correct extensions`, () => {
    expect(resolvableExtensions()).toMatchSnapshot()
  })

  it(`modifies webpack config`, () => {
    const config = {
      loader: jest.fn(),
    }

    modifyWebpackConfig({ config }, { compilerOptions: {} })

    expect(config.loader).toHaveBeenCalledTimes(1)
    const lastCall = config.loader.mock.calls.pop()
    expect(lastCall).toMatchSnapshot()
  })

  describe(`pre-processing`, () => {
    const opts = { compilerOptions: {} }
    it(`leaves non-tsx? files alone`, () => {
      expect(
        preprocessSource(
          {
            contents: `alert('hello');`,
            filename: `test.js`,
          },
          opts
        )
      ).toBeNull()
    })

    it(`transforms .ts files`, () => {
      const js = preprocessSource(
        {
          filename: `index.ts`,
          contents: `
          declare let moment: any;

          const now: string = moment().format('HH:MM:ss');
        `,
        },
        opts
      )
      expect(js).not.toBeNull()
      expect(js).toMatchSnapshot()
    })

    it(`transforms JSX files`, () => {
      const js = preprocessSource(
        {
          filename: `tags.ts`,
          contents: `
          import * as React from 'react';

          export default () => <h1>Hello World</h1>;
        `,
        },
        opts
      )

      expect(js).not.toBeNull()
      expect(js).toMatchSnapshot()
    })
  })
})
