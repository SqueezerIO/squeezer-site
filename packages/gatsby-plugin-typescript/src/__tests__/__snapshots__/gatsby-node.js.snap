// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gatsby-plugin-typescript modifies webpack config 1`] = `
Array [
  "typescript",
  Object {
    "loaders": Array [
      "babel?{\\"plugins\\":[\\"\\"]}",
      "ts-loader?{\\"compilerOptions\\":{\\"target\\":\\"esnext\\",\\"experimentalDecorators\\":true,\\"jsx\\":\\"react\\",\\"module\\":\\"commonjs\\"},\\"transpileOnly\\":true}",
    ],
    "test": /\\\\\\.tsx\\?\\$/,
  },
]
`;

exports[`gatsby-plugin-typescript pre-processing transforms .ts files 1`] = `
"const now = moment().format('HH:MM:ss');
"
`;

exports[`gatsby-plugin-typescript pre-processing transforms JSX files 1`] = `
"import * as React from 'react';
export default () => React.createElement(\\"h1\\", null, \\"Hello World\\");
"
`;

exports[`gatsby-plugin-typescript returns correct extensions 1`] = `
Array [
  ".ts",
  ".tsx",
]
`;
